docs/knowledgebase/advanced/executor.md

```diff
@@ -20,53 +20,51 @@ Before runtime execution begins, the Substrate client proposes which runtime exe
 should be used. This is controlled by the execution strategy, which can be configured for the
 different parts of the blockchain execution process. The strategies are:
 
-- `NativeWhenPossible`: Execute with the native equivalent if it is compatible with the given
-wasm module; otherwise fall back to the wasm.
-- `Wasm`: Use the given wasm module..
-- `Both`: Run with both the wasm and the native variant (if compatible).
-Report any discrepancy as an error.
-- `NativeElseWasm`: First native, then if that fails or is not possible, wasm.
+- `NativeWhenPossible`: Execute with native build (if available, WebAssembly otherwise).
```

Comment:
I do not see these options on the node template:

```bash
# node template
./target/release/node-template --dev --tmp --execution=asd
error: 'asd' isn't a valid value for '--execution <STRATEGY>'
	[possible values: Both, Native, NativeElseWasm, Wasm]
	
	#substrate full bin build
	./target/release/substrate --dev --tmp --execution=asd
error: 'asd' isn't a valid value for '--execution <STRATEGY>'
	[possible values: Both, Native, NativeElseWasm, Wasm]
	```
	
Are these options incorrect in the docs or does the node CLI need an update (or both)?

Reply:
I think this is the equivalent of just `Native` from the CLI.

Reply:
Right. I think the docs have a deprecated set of options... and what is the difference between `Native` and `NativeIfPossible` ?

Reply:
I assume that "when possible" (was) trying to get at the fact that WASM will be used if the runtime is upgraded? Is that still the case, or does `Native` force use of native, and ignores wasm?

Reply:
The confusion arises because the CLI argument is called one thing and in `sc_client_api` it is called something similar (but not necessarily the same), and there is an underlying one-to-one mapping, [shown here](https://github.com/paritytech/substrate/blob/9b08105b8c/client/cli/src/arg_enums.rs#L120-L129)

- cli arg `Native` maps to `sc_client_api` `NativeWhenPossible`
- cli arg `Wasm` maps to `sc_client_api` `AlwaysWasm`
- cli arg `Both` maps to `sc_client_api` `Both`
- cli arg `NativeElseWasm ` maps to `sc_client_api` `NativeElseWasm`

I am using the `sc_client_api` naming in the doc.

Reply:
@kianenigma There are some questions which is along the line of Dan's one. I understand executor will check the runtime version of the native and wasm runtime. If the native runtime version is the same as wasm's one, then `NativeWhenPossible` and `NativeElseWasm` strategy will choose native runtime.

But if not, and the wasm runtime version is more updated, is `NativeWhenPossible` choosing the wasm runtime? In this case, which runtime does `NativeElseWasm` strategy pick?

In what scenario do we want to use the `Both` strategy (running with both runtime)? And I assume this only application if both runtime version is the same. If native & wasm runtime version is different, there is no point in running `Both` as it will always have difference and cause error.